apps/client/src/module/appsetting/language-switcher.ts

```diff
@@ -0,0 +1,47 @@
+"use client";
+
+import { getNextLocale } from "@src/actions";
+import { useEffect, useState } from "react";
+
+type locales = "en" | "zh-cn";
+
+export function useLanguageSwitcher() {
+  const [currentLocale, setCurrentLocale] = useState<locales>("en");
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getNextLocale().then((locale) => {
+      if (isLocale(locale)) {
+        setCurrentLocale(locale);
+        setLoading(false);
+      }
+    });
+  }, []);
+
+  // Locales switch function and rules
+  const localesSwitchRules = {
+    English: "en",
+    中文: "zh-cn",
+    en: "English",
+    "zh-cn": "中文",
+  };
+  const optionalLocales = ["en", "zh-cn"];
+
+  const switchLanguageValue = (
+    lang: keyof typeof localesSwitchRules,
+  ): string => {
+    return localesSwitchRules[lang];
```

Comment:
so what does this function do?

Reply:
changed the data structure as you mentored